internal/pkg/cli/task_delete.go

```diff
@@ -262,26 +290,131 @@ func (o *deleteTaskOpts) askTaskName() error {
 		if err != nil {
 			return fmt.Errorf("select task from default cluster: %w", err)
 		}
-		o.name = task.Name
+		o.name = task
 		return nil
 	}
 	task, err := sel.Task(taskDeleteNamePrompt, "", selector.TaskWithAppEnv(o.app, o.env))
 	if err != nil {
 		return fmt.Errorf("select task from environment: %w", err)
 	}
-	o.name = task.Name
+	o.name = task
 	return nil
 }
 
 func (o *deleteTaskOpts) Execute() error {
-	// Get clients.
-
-	// Stop tasks.
+	// Stop all running tasks in given family.
+	if err := o.stopTasks(); err != nil {
+		return err
+	}
 
 	// Clear repository.
+	if err := o.clearECRRepository(); err != nil {
+		return err
+	}
 
 	// Delete stack.
+	if err := o.deleteStack(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *deleteTaskOpts) stopTasks() error {
+	sess, err := o.getSession()
+	if err != nil {
+		return fmt.Errorf("get session: %w", err)
+	}
+
+	o.spinner.Start(fmt.Sprintf("Stopping all running tasks in family %s.", color.HighlightUserInput(o.name)))
+
+	// Stop tasks.
+	if o.defaultCluster {
+		if err = o.newTaskStopper(sess).StopDefaultClusterTasks(o.name); err != nil {
+			o.spinner.Stop(log.Serrorln("Error stopping running tasks in default cluster."))
+			return fmt.Errorf("stop running tasks in family %s: %w", o.name, err)
+		}
+	} else {
+		if err = o.newTaskStopper(sess).StopAppEnvOneOffTasks(o.app, o.env, o.name); err != nil {
+			o.spinner.Stop(log.Serrorln("Error stopping running tasks in environment."))
+			return fmt.Errorf("stop running tasks in family %s: %w", o.name, err)
+		}
+	}
+	o.spinner.Stop(log.Ssuccessf("Stopped all running tasks in family %s.\n", color.HighlightUserInput(o.name)))
+	return nil
+}
+
+func (o *deleteTaskOpts) clearECRRepository() error {
+	// ECR Deletion happens from the default profile in app delete. We can do it here too by getting
+	// a default session in whichever region we're deleting from.
+	var defaultSess *awssession.Session
+	var err error
+	if o.defaultCluster {
+		defaultSess, err = o.sess.Default()
```

Comment:
Is `o.sess.Default()` equal to `o.getSession()` when `o.defaultCluster` is true?

Reply:
yes, and I've made this change.